### PR TITLE
Clock compensation refactoring

### PR DESCRIPTION
**Pull request by Eitu33**
closes https://github.com/massalabs/massa/issues/3245

### CODE REVIEW

File: massa-execution-worker/src/tests/scenarios_mandatories.rs [hunk 1]
```diff
@@ -724,6 +724,8 @@ pub fn roll_buy() {
 
 #[test]
 #[serial]
+// IMPORTANT TODO: find out what is causing this, not coming from this update
```
Comment (sydhds) [inline massa-execution-worker/src/tests/scenarios_mandatories.rs:727]: Should link to an issue here as it's too important for testnet 18

File: massa-node/base_config/config.toml [hunk 2]
```diff
@@ -211,8 +211,8 @@
     read_error_timeout = 200
     # timeout for message error sending
     write_error_timeout = 200
-    # when enabled, apply a correction to the local computer clock to match bootstrap server time
-    enable_clock_synchronization = false
+    # max allowed difference between client and servers clocks
```
Comment (sydhds) [inline massa-node/base_config/config.toml:214]: Add unit here?

File: massa-client/src/tests/tools.rs [hunk 3]
```diff
@@ -7,9 +7,8 @@ use toml_edit::{value, Document};
 pub fn _update_genesis_timestamp(config_path: &str) {
     let toml = fs::read_to_string(config_path).expect("Unable to read file");
     let mut doc = toml.parse::<Document>().unwrap();
-    doc["consensus"]["genesis_timestamp"] = value(format!(
-        "{}",
-        MassaTime::now(10000 * 60 * 60).unwrap().to_millis()
-    ));
+    // IMPORTANT TODO: might be sus
```
Comment (sydhds) [inline massa-client/src/tests/tools.rs:10]: sus?

File: massa-bootstrap/src/client.rs [hunk 4]
```diff
@@ -267,39 +267,30 @@ async fn bootstrap_from_server(
...
-        let compensation_millis: i64 = compensation_millis.try_into().map_err(|_| {
-            BootstrapError::GeneralError("Failed to convert compensation time into i64".into())
-        })?;
-        debug!("Server clock compensation set to: {}", compensation_millis);
-        compensation_millis
-    } else {
-        0
-    };
+    // compute client / server clock delta
+    let adjusted_server_time = server_time.checked_add(ping.checked_div_u64(2)?)?;
```
Comment (sydhds) [inline massa-bootstrap/src/client.rs:282]: Why the divide by 2 here?

**Review by sydhds — CHANGES_REQUESTED** [review]
NIce PR. A lot of code removed :)

File: massa-bootstrap/src/client.rs (hunk 4 above)
Comment (Eitu33) [reply massa-bootstrap/src/client.rs:282]: Approximation of the time it took the message to do `server -> client`, the complete ping value being `client -> server -> client`

Comment (sydhds) [reply massa-bootstrap/src/client.rs:282]: Maybe you can add this explanation in the comment ?

**Review by AurelienFT — APPROVED** [review]

**Pull request merged**